File: app/src/main/java/com/ne1c/developerstalk/models/view/RoomViewModel.java
package com.ne1c.developerstalk.models.view;

import android.os.Parcel;
import android.os.Parcelable;

public class RoomViewModel implements Parcelable {
    public String id;
    public String name;
    public String topic;
    public boolean oneToOne;
    public int unreadItems;
    public int userCount;
    public boolean mention;
    public boolean hide;
    public int listPosition = -1;

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.id);
        dest.writeString(this.name);
        dest.writeString(this.topic);
        dest.writeByte(oneToOne ? (byte) 1 : (byte) 0);
        dest.writeInt(this.unreadItems);
        dest.writeInt(this.userCount);
        dest.writeByte(mention ? (byte) 1 : (byte) 0);
        dest.writeByte(hide ? (byte) 1 : (byte) 0);
        dest.writeInt(this.listPosition);
    }

    public RoomViewModel() {
    }

    protected RoomViewModel(Parcel in) {
        this.id = in.readString();
        this.name = in.readString();
        this.topic = in.readString();
        this.oneToOne = in.readByte() != 0;
        this.unreadItems = in.readInt();
        this.userCount = in.readInt();
        this.mention = in.readByte() != 0;
        this.hide = in.readByte() != 0;
        this.listPosition = in.readInt();
    }

    public static final Parcelable.Creator<RoomViewModel> CREATOR = new Parcelable.Creator<RoomViewModel>() {
        @Override
        public RoomViewModel createFromParcel(Parcel source) {
            return new RoomViewModel(source);
        }

        @Override
        public RoomViewModel[] newArray(int size) {
            return new RoomViewModel[size];
        }
    };
}

File: app/src/main/kotlin/com/ne1c/gitteroid/models/view/RoomViewModel.kt
package com.ne1c.gitteroid.models.view

import android.os.Parcel
import android.os.Parcelable

data class RoomViewModel(
        var id: String = "",
        var name: String = "",
        var topic: String = "",
        var oneToOne: Boolean = false,
        var unreadItems: Int = 0,
        var userCount: Int = 0,
        var mention: Boolean = false,
        var hide: Boolean = false,
        var listPosition: Int = -1) : Parcelable {
    fun getAvatarUrl(): String {
        if (name.contains("/")) {
            return "https://avatars.githubusercontent.com/" + name.substring(0, name.indexOf("/"))
        } else {
            return "https://avatars.githubusercontent.com/" + name
        }
    }

    constructor(source: Parcel) : this(source.readString(),
            source.readString(),
            source.readString(),
            1.equals(source.readInt()),
            source.readInt(),
            source.readInt(),
            1.equals(source.readInt()),
            1.equals(source.readInt()),
            source.readInt())

    override fun describeContents() = 0

    override fun writeToParcel(dest: Parcel?, flags: Int) {
        dest?.writeString(id)
        dest?.writeString(name)
        dest?.writeString(topic)
        dest?.writeInt((if (oneToOne) 1 else 0))
        dest?.writeInt(unreadItems)
        dest?.writeInt(userCount)
        dest?.writeInt((if (mention) 1 else 0))
        dest?.writeInt((if (hide) 1 else 0))
        dest?.writeInt(listPosition)
    }

    companion object {
        @JvmField val CREATOR: Parcelable.Creator<RoomViewModel> = object : Parcelable.Creator<RoomViewModel> {
            override fun createFromParcel(source: Parcel): RoomViewModel = RoomViewModel(source)
            override fun newArray(size: Int): Array<RoomViewModel?> = arrayOfNulls(size)
        }
    }
}
